web3/contract/contract.py

```diff
@@ -245,28 +255,55 @@ def __init__(
     ) -> None:
         super().__init__(abi, w3, ContractEvent, address)
 
+    def __getattr__(self, event_name: str) -> "ContractEvent":
+        if super().__getattribute__("abi") is None:
+            raise NoABIFound(
+                "There is no ABI found for this contract.",
+            )
+        if "_events" not in self.__dict__:
+            raise NoABIEventsFound(
+                "The abi for this contract contains no event definitions. ",
+                "Are you sure you provided the correct contract abi?",
+            )
+        elif get_name_from_abi_element_identifier(event_name) not in [
+            get_name_from_abi_element_identifier(event["name"])
+            for event in self._events
+        ]:
+            raise ABIEventNotFound(
+                f"The event '{event_name}' was not found in this contract's abi. ",
+                "Are you sure you provided the correct contract abi?",
+            )
+        else:
+            event_abi = get_abi_element(self._events, event_name)
+            argument_types = get_abi_input_types(event_abi)
+            event_signature = str(get_abi_element_signature(event_name, argument_types))
+            return super().__getattribute__(event_signature)
+
+    def __getitem__(self, event_name: str) -> "ContractEvent":
+        return getattr(self, event_name)
+
+    def __iter__(self) -> Iterable["ContractEvent"]:
+        """
+        Iterate over supported
```

Comment:
nit - docstring should state what the function does (not sure if it's necessary for a magic method?). Also should match for the same method in `AsyncContractEvents`.

Reply:
Good point, there's no practical information I provided there anyway. Removing!